resources/helm/dask-gateway/templates/gateway/deployment.yaml

```diff
@@ -57,3 +57,19 @@ spec:
           ports:
             - containerPort: 8000
               name: api
+          livenessProbe:
+            httpGet:
+              path: /api/health
+              port: api
+            initialDelaySeconds: 5
+            periodSeconds: 10
+            timeoutSeconds: 2
+            failureThreshold: 3
+          readinessProbe:
+            httpGet:
+              path: /api/health
+              port: api
+            initialDelaySeconds: 5
+            periodSeconds: 10
+            timeoutSeconds: 2
+            failureThreshold: 1
```

Comment:
The `readinessProbe` fails first, which cuts traffic to the impacted pod but doesn't restart it, giving the application time to recover. After 3 failed health checks (line 67), the recovery window closes, the `livenessProbe` is triggered, and the pod is replaced.